### PR TITLE
types: refactor multiple fuzzers

### PR DESCRIPTION
**Pull request by AdamKorcz**
Signed-off-by: AdamKorcz <adam@adalogics.com>

#### Summary

Adds a fuzzer for intoto v0.0.2

Does the following changes to multiple types fuzzers:

1. Silences the logger.
2. Creates the artifact file if the artifact bytes are `nil` for the properties.
3. Hardcodes the version to avoid unnecessary time in the semver dependency.
4. Moves the fuzzers to their v0.0.1 directory to initiate the relevant entry in the VersionMap. For example here:
https://github.com/sigstore/rekor/blob/26f44cd4ba51119da6b9557edcadabe3a97ae2af/pkg/types/helm/v0.0.1/entry.go#L46-L50

### CODE REVIEW

**Review by bobcallaway — DISMISSED** [review]

File: pkg/types/cose/fuzz_test.go [hunk 1]
```diff
@@ -17,13 +17,18 @@ package cose
...
 
+	fuzzUtils "github.com/sigstore/rekor/pkg/fuzz"
 	"github.com/sigstore/rekor/pkg/types"
 )
 
+var initter sync.Once
+
 func FuzzCreateProposedEntry(f *testing.F) {
 	f.Fuzz(func(t *testing.T, version string) {
+		initter.Do(fuzzUtils.SetFuzzLogger)
```
Comment (asraa) [inline pkg/types/cose/fuzz_test.go:31]: should this be moved to cose v0.0.1 and populate the props with `CreateProps` like other types?

File: pkg/fuzz/fuzz_utils.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+		artifactPath, err := filepath.Abs("ArtifactFile")
+		if err != nil {
+			return props, nil, err
+		}
+		artifactURL, err := url.Parse(artifactPath)
+		if err != nil {
+			return props, nil, err
+		}
+		props.ArtifactPath = artifactURL
```
Comment (asraa) [inline pkg/fuzz/fuzz_utils.go:55]: In most cases, if `ArtifactBytes` are written, then we won't even look at the `ArtifactPath` (https://github.com/sigstore/rekor/blob/8fbe9e4a5424c8e8ea2fe668848526202fa46d87/pkg/types/rekord/v0.0.1/entry.go#L343-L345) -- maybe instead you can take some input from `ff` to either use `ArtifactBytes` directly or use it from `ArtifactPath` and clear `ArtifactBytes`?

The same goes for `SignatureBytes` and `SignaturePath`/`PublicKeyBytes` and `PublicKeyPaths`

**Review by asraa — APPROVED** [review]
Thanks! Feel free to address the review in a follow-up, they're minor.

File: pkg/fuzz/fuzz_utils.go (hunk 2 above)
Comment (asraa) [reply pkg/fuzz/fuzz_utils.go:55]: I misread this! If the ArtifactBytes are nil -- lgtm! (Just see about the Signature/PublicKey if there are any issues)

**Pull request merged**